### PR TITLE
MDEP-763 - Minor Improvements:

### PR DESCRIPTION
**Pull request by arturobernalg**
* Inline variable
* Avoid explicit
* Redundant 'Collection.addAll()' call
* Make final variables

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [X ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ X] Each commit in the pull request should have a meaningful subject line and body.
 - [X ] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [X ] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java [hunk 1]
```diff
@@ -325,7 +325,7 @@ protected void unpack( Artifact artifact, String type, File location, String inc
         catch ( ArchiverException e )
         {
             throw new MojoExecutionException( "Error unpacking file: " + file + " to: " + location
-                + System.lineSeparator() + e.toString(), e );
+                + System.lineSeparator() + e, e );
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java:328]: I consider this pointless to incude `e.toString()` in the exception message. One already passed `e`. That is more than enough.

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/BuildClasspathMojo.java [hunk 2]
```diff
@@ -347,7 +347,7 @@ private void storeClasspathFile( String cpString, File out )
         }
         catch ( IOException ex )
         {
-            throw new MojoExecutionException( "Error while writing to classpath file '" + out + "': " + ex.toString(),
+            throw new MojoExecutionException( "Error while writing to classpath file '" + out + "': " + ex,
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/fromDependencies/BuildClasspathMojo.java:350]: Same here.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java]: HI @michael-o
Agree. Removed.
TY

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/BuildClasspathMojo.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/maven/plugins/dependency/fromDependencies/BuildClasspathMojo.java]: idem above.

File: src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java [hunk 3]
```diff
@@ -182,7 +182,7 @@
      * @since 2.10
      */
     @Parameter
-    private String[] ignoredDependencies = new String[0];
+    private final String[] ignoredDependencies = new String[0];
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:185]: What is benefit with final variable, which is used as Mojo parameters? 

Those fields will be injected by maven/plexus so reflections on final fields is more complicated - today it is works ...

**Pull request closed without merging**